--- NetworkManager-1.8.0/configure.ac.0003~	2017-05-11 15:30:17.967321401 +0200
+++ NetworkManager-1.8.0/configure.ac	2017-05-11 15:31:00.459597432 +0200
@@ -36,8 +36,10 @@ AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
 # Prefer gcc-* variants; the ones libtool would choose don't work with LTO
-AC_CHECK_TOOLS(AR, [gcc-ar ar], false)
-AC_CHECK_TOOLS(RANLIB, [gcc-ranlib ranlib], :)
+# Unless we're on OpenMandriva, which actually fixes things instead of breaking them
+# gcc-ar and gcc-ranlib break clang lto
+#AC_CHECK_TOOLS(AR, [gcc-ar ar], false)
+#AC_CHECK_TOOLS(RANLIB, [gcc-ranlib ranlib], :)
 
 dnl Initialize libtool
 LT_PREREQ([2.2])
