diff -up NetworkManager-1.12.0/data/NetworkManager.service.in.0051~ NetworkManager-1.12.0/data/NetworkManager.service.in
--- NetworkManager-1.12.0/data/NetworkManager.service.in.0051~	2018-06-30 11:46:03.137159190 +0200
+++ NetworkManager-1.12.0/data/NetworkManager.service.in	2018-06-30 18:40:05.040519953 +0200
@@ -25,6 +25,7 @@ ProtectHome=read-only
 [Install]
 WantedBy=multi-user.target
 Alias=dbus-org.freedesktop.NetworkManager.service
+Alias=network-manager.service
 Also=NetworkManager-dispatcher.service
 
 # We want to enable NetworkManager-wait-online.service whenever this service
