diff -up NetworkManager-1.26.0/meson.build.omv~ NetworkManager-1.26.0/meson.build
--- NetworkManager-1.26.0/meson.build.omv~	2020-08-11 04:05:50.451540422 +0200
+++ NetworkManager-1.26.0/meson.build	2020-08-11 04:06:23.771068428 +0200
@@ -290,7 +290,7 @@ glib_dep = declare_dependency(
   ]
 )
 
-if run_command('test', '-e', '/etc/sysconfig/network-scripts').returncode() == 0
+if run_command('test', '-e', '/etc/sysconfig/rh-network-scripts').returncode() == 0
   distro = 'redhat'
 elif run_command('test', '-e', '/etc/SuSE-release').returncode() == 0
   distro = 'suse'
@@ -299,7 +299,7 @@ elif run_command('test', '-e', '/etc/deb
 elif run_command('test', '-e', '/etc/gentoo-release').returncode() == 0
   distro = 'gentoo'
 else
-  distro = 'unknown'
+  distro = 'openmandriva'
 endif
 
 enable_ifcfg_rh = get_option('ifcfg_rh') or (distro == 'redhat')
