diff -uNr NetworkManager-0.9.4.0/libnm-glib/nm-remote-settings.c NetworkManager-0.9.4.0p/libnm-glib/nm-remote-settings.c
--- NetworkManager-0.9.4.0/libnm-glib/nm-remote-settings.c	2012-03-21 19:31:38.000000000 +0200
+++ NetworkManager-0.9.4.0p/libnm-glib/nm-remote-settings.c	2012-04-02 13:59:00.949490963 +0300
@@ -728,7 +728,12 @@
 NMRemoteSettings *
 nm_remote_settings_new (DBusGConnection *bus)
 {
-	return g_object_new (NM_TYPE_REMOTE_SETTINGS, NM_REMOTE_SETTINGS_BUS, bus, NULL);
+	NMRemoteSettings *settings;
+
+	settings = g_object_new (NM_TYPE_REMOTE_SETTINGS, NM_REMOTE_SETTINGS_BUS, bus, NULL);
+
+	_nm_remote_settings_ensure_inited (settings);
+	return settings;
 }
 
 static void
