diff -up NetworkManager-1.40.0/meson.build.2~ NetworkManager-1.40.0/meson.build
--- NetworkManager-1.40.0/meson.build.2~	2022-08-27 23:56:09.688622571 +0200
+++ NetworkManager-1.40.0/meson.build	2022-08-27 23:57:17.735640226 +0200
@@ -289,7 +289,7 @@ glib_dep = declare_dependency(
 
 enable_ifcfg_rh = get_option('ifcfg_rh')
 if enable_ifcfg_rh == 'auto'
-  enable_ifcfg_rh = (run_command('test', '-e', '/etc/sysconfig/network-scripts').returncode() == 0)
+  enable_ifcfg_rh = (run_command('test', '-e', '/etc/sysconfig/rh-network-scripts').returncode() == 0)
 else
   enable_ifcfg_rh = (enable_ifcfg_rh != 'false')
 endif
