diff -Naur NetworkManager-0.9.10.2/src/dhcp-manager/nm-dhcp-dhcpcd.c NetworkManager-0.9.10.2.tpg/src/dhcp-manager/nm-dhcp-dhcpcd.c
--- NetworkManager-0.9.10.2/src/dhcp-manager/nm-dhcp-dhcpcd.c	2015-03-29 10:29:22.000000000 +0000
+++ NetworkManager-0.9.10.2.tpg/src/dhcp-manager/nm-dhcp-dhcpcd.c	2015-03-29 10:34:22.804017014 +0000
@@ -134,6 +134,8 @@
 
 	g_ptr_array_add (argv, (gpointer) "-G");	/* Let NM handle routing */
 
+	g_ptr_array_add (argv, (gpointer) "-4");        /* Only handle the new IPv4 support for now, otherwise we'll have to implement it in the settings plugins as well.. :p */
+
 	g_ptr_array_add (argv, (gpointer) "-c");	/* Set script file */
 	g_ptr_array_add (argv, (gpointer) nm_dhcp_helper_path);
 
