--- NetworkManager-1.9.90/configure.ac.0003~	2017-11-06 21:52:04.607862222 +0100
+++ NetworkManager-1.9.90/configure.ac	2017-11-06 22:03:30.622032009 +0100
@@ -36,9 +36,11 @@ AC_PROG_LN_S
 AC_PROG_MKDIR_P
 
 # Prefer gcc-* variants; the ones libtool would choose don't work with LTO
-AC_CHECK_TOOLS(AR, [gcc-ar ar], false)
-AC_CHECK_TOOLS(RANLIB, [gcc-ranlib ranlib], :)
-AC_CHECK_TOOLS(NM, [$BINUTILS_NM gcc-nm nm])
+#AC_CHECK_TOOLS(AR, [gcc-ar ar], false)
+#AC_CHECK_TOOLS(RANLIB, [gcc-ranlib ranlib], :)
+#AC_CHECK_TOOLS(NM, [$BINUTILS_NM gcc-nm nm])
+# Unless we're on OpenMandriva, which actually fixes things instead of breaking them
+# gcc-ar and gcc-ranlib break clang lto
 
 dnl Initialize libtool
 LT_PREREQ([2.2])
