diff -up NetworkManager-1.16.0/data/NetworkManager.service.in.3~ NetworkManager-1.16.0/data/NetworkManager.service.in
--- NetworkManager-1.16.0/data/NetworkManager.service.in.3~	2019-03-16 23:30:33.878478228 +0100
+++ NetworkManager-1.16.0/data/NetworkManager.service.in	2019-03-16 23:31:22.893400159 +0100
@@ -25,6 +25,7 @@ ProtectHome=read-only
 [Install]
 WantedBy=multi-user.target
 Also=NetworkManager-dispatcher.service
+Alias=network-manager.service
 
 # We want to enable NetworkManager-wait-online.service whenever this service
 # is enabled. NetworkManager-wait-online.service has
