--- NetworkManager-0.9.4.0/src/backends/NetworkManagerMandriva.c.nscd~	2012-05-12 18:00:25.206634303 +0200
+++ NetworkManager-0.9.4.0/src/backends/NetworkManagerMandriva.c	2012-05-12 18:00:26.901644239 +0200
@@ -37,7 +37,11 @@ void nm_backend_update_dns (void)
 {
 	/* Invalidate the nscd host cache, if it exists, since we changed resolv.conf */
 	if (g_file_test ("/usr/sbin/nscd", G_FILE_TEST_EXISTS | G_FILE_TEST_IS_EXECUTABLE | G_FILE_TEST_IS_REGULAR)) {
-		nm_spawn_process ("/etc/init.d/nscd condrestart");
+		if (g_file_test ("/etc/init.d/nscd", G_FILE_TEST_EXISTS))
+			nm_spawn_process ("/etc/init.d/nscd condrestart");
+		else if (g_file_test ("/bin/systemctl", G_FILE_TEST_IS_EXECUTABLE))
+			nm_spawn_process ("/bin/systemctl condrestart nscd.service");
+
 		nm_spawn_process ("/usr/sbin/nscd -i hosts");
 	}
 }
