diff -Naur NetworkManager-0.9.10.2/policy/org.freedesktop.NetworkManager.policy.in.in NetworkManager-0.9.10.2.tpg/policy/org.freedesktop.NetworkManager.policy.in.in
--- NetworkManager-0.9.10.2/policy/org.freedesktop.NetworkManager.policy.in.in	2014-12-09 21:17:15.000000000 +0000
+++ NetworkManager-0.9.10.2.tpg/policy/org.freedesktop.NetworkManager.policy.in.in	2015-03-29 10:19:46.055328679 +0000
@@ -98,7 +98,7 @@
     <defaults>
       <allow_any>auth_admin_keep</allow_any>
       <allow_inactive>@NM_MODIFY_SYSTEM_POLICY@</allow_inactive>
-      <allow_active>@NM_MODIFY_SYSTEM_POLICY@</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
