--- a/policy/org.freedesktop.NetworkManager.policy.in.in	2013-02-21 00:56:16.000000000 +0400
+++ b/policy/org.freedesktop.NetworkManager.policy.in.in	2013-02-26 09:22:11.487851110 +0400
@@ -95,7 +95,7 @@
     <_message>System policy prevents modification of network settings for all users</_message>
     <defaults>
       <allow_inactive>no</allow_inactive>
-      <allow_active>@NM_MODIFY_SYSTEM_POLICY@</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
